dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -1275,4 +1342,42 @@ private void CaptureUsageDetails(CompletionsUsage usage)
         return JsonSerializer.Serialize(functionResult, toolCallBehavior?.ToolCallResultSerializerOptions);
 #pragma warning restore CS0618 // Type or member is obsolete
     }
+
+    /// <summary>
+    /// Executes auto function invocation filters and/or function itself.
+    /// This method can be moved to <see cref="Kernel"/> when auto function invocation logic will be extracted to common place.
+    /// </summary>
+    private static async Task<AutoFunctionInvocationContext> OnAutoFunctionInvocationAsync(
```

Comment:
This method simply calls another one, `InvokeFilterOrFunctionAsync`. Can they be merged into one instead?

Reply:
Method `InvokeFilterOrFunctionAsync` is invoked in recursion manner, and in `OnAutoFunctionInvocationAsync` it is executed with `0` index, as a default parameter. In my initial version I kept this method as a local one, but then I decided to move it to separate method for better readability.